test/e2e_node/eviction_test.go

```diff
@@ -520,19 +520,34 @@ func runEvictionTest(f *framework.Framework, pressureTimeout time.Duration, expe
 		})
 
 		ginkgo.AfterEach(func() {
-			defer func() {
+			prePullImagesIfNeccecary := func() {
 				if expectedNodeCondition == v1.NodeDiskPressure && framework.TestContext.PrepullImages {
 					// The disk eviction test may cause the prepulled images to be evicted,
 					// prepull those images again to ensure this test not affect following tests.
 					PrePullAllImages()
 				}
-			}()
+			}
+			defer prePullImagesIfNeccecary()
```

Comment:
Can you share more about the purpose of the `defer` statement given the other changes you made?

Reply:
No problem! :smile: 

ref. #82018, it makes sense to use defer since we want to run the pre pulling even tho the assertions in `AfterEach` fails.

After we merged #82018, line [553](https://github.com/kubernetes/kubernetes/pull/82654/files#diff-590eb0d04c6b4987c1addfa5d2c4424aR553) started failing, since the image is used there can also be evicted too. I think we should do prepulling in `defer` as well as before the `PodClient().CreateSync`, so that we are 100% sure all images are present.

Reply:
Works for me :) Might be worth copying this over into a comment (or putting in the commit message).

Reply:
Added a comment now :smile: (but lost the lgtm :unamused:)